rosbag2/src/rosbag2/writer.cpp

```diff
@@ -25,10 +23,30 @@
 
 #include "rosbag2/info.hpp"
 #include "rosbag2/storage_options.hpp"
+#include "rosbag2_storage/filesystem_helper.hpp"
 
 namespace rosbag2
 {
 
+namespace
+{
+std::string format_storage_uri(const std::string & base_folder, uint64_t storage_count)
+{
+  // Right now `base_folder_` is always just the folder name for where to install the bagfile.
+  // The name of the folder needs to be queried in case Writer is opened with a relative path.
+  std::stringstream storage_file_name;
+  storage_file_name << rosbag2_storage::FilesystemHelper::get_folder_name(base_folder);
+
+  // Only append the counter after we have split.
+  // This is so bagfiles have the old naming convention if splitting is disabled.
```

Comment:
I was talking with @piraka9011 about this last week, part of the reason for supporting the old convention is so the codebase doesn't break if this merges in before the Reader changes.

Reply:
why would that be the case? The relative file path doesn't follow any naming conventions at the moment. The only requirement is that it points to a valid file path.

Reply:
I thought about this one a bit more and I think the only way it would break anything is if something is making assumptions on what the bagfiles are named instead of reading the metadata. Reading the bagfile relative path from the metadata should be the correct way to retrieve the bagfiles.

I'll change this not have the branch